### PR TITLE
Fix "cannot read property 'title' of undefined" after HMR

### PR DESCRIPTION
**Pull request by BinaryMuse**
Something about the way HMR runs is causing every build except the first one to have a missing `frontmatter` property in the page context; this PR adds the frontmatter explicitly, ensuring it always exists.

This PR also fine tunes the way we look up contributor data from GitHub:

* If `GITHUB_TOKEN` is not set, no API call will be made
* Results from the API are stored in an in-memory cache to cut down on requests during development

### CODE REVIEW

**Comment by BinaryMuse** [comment]
This fixes the issues I was complaining about in Slack.

File: theme/gatsby-node.js [hunk 1]
```diff
@@ -68,11 +82,23 @@ function getEditUrl(repo, filePath) {
...
+  }
+
   try {
-    const {data} = await axios.get(
-      `https://api.github.com/repos/${repo.user}/${repo.project}/commits?path=${filePath}`,
-    )
+    const {data} = await axios.request({
+      method: 'get',
+      baseURL: 'https://api.github.com/',
+      url: `/repos/${repo.user}/${repo.project}/commits?path=${filePath}&per_page=100`,
```
Comment (colebemis) [inline theme/gatsby-node.js]: Good call setting `per_page` to 100 and using an access token 👍 We were probably missing contributors before 😬

File: theme/gatsby-node.js [hunk 2]
```diff
@@ -68,11 +82,23 @@ function getEditUrl(repo, filePath) {
   return `https://github.com/${repo.user}/${repo.project}/edit/master/${filePath}`
 }
 
-async function fetchContributors(repo, filePath) {
+async function fetchContributors(repo, filePath, accessToken) {
+  const hash = `${repo.user}/${repo.project}/${filePath}`
+  const cached = CONTRIBUTOR_CACHE.get(hash)
+  if (cached) {
+    console.debug('found something in the cache. whee!')
```
Comment (colebemis) [inline theme/gatsby-node.js]: [Minor] I'd say we should use sentence case for consistency.

```suggestion
    console.debug('Found something in the cache. Whee!')
```

Comment (colebemis) [reply theme/gatsby-node.js]: Oh, looks like you took care of this already 👍

**Pull request merged**